packages/datadog-plugin-aws-sdk/test/dynamodb.spec.js

```diff
@@ -0,0 +1,878 @@
+          action: { S: 'login' }
+        }
+        const keyConfig = { UserActivityTable: new Set(['userId', 'timestamp']) }
+
+        const actualHash = DynamoDb.calculatePutItemHash(tableName, item, keyConfig)
+        const expectedHash = generatePointerHash([tableName, 'timestamp', '1234567', 'userId', 'user123'])
+        expect(actualHash).to.equal(expectedHash)
+      })
+
+      it('generates correct hash for binary-binary key', () => {
+        const tableName = 'BinaryTable'
+        const binary1 = Buffer.from('abc')
+        const binary2 = Buffer.from('1ef230')
+        const item = {
+          key1: { B: binary1 },
+          key2: { B: binary2 },
+          data: { S: 'test' }
+        }
+        const keyConfig = { BinaryTable: new Set(['key1', 'key2']) }
+
+        const actualHash = DynamoDb.calculatePutItemHash(tableName, item, keyConfig)
+        const expectedHash = generatePointerHash([tableName, 'key1', binary1, 'key2', binary2])
+        expect(actualHash).to.equal(expectedHash)
+      })
+
+      it('generates unique hashes for different tables', () => {
+        const item = { userId: { S: 'user123' } }
+        const keyConfig = {
+          Table1: new Set(['userId']),
+          Table2: new Set(['userId'])
+        }
+
+        const hash1 = DynamoDb.calculatePutItemHash('Table1', item, keyConfig)
+        const hash2 = DynamoDb.calculatePutItemHash('Table2', item, keyConfig)
+        expect(hash1).to.not.equal(hash2)
+      })
+
+      describe('edge cases', () => {
+        it('returns undefined for unknown table', () => {
+          const tableName = 'UnknownTable'
+          const item = { userId: { S: 'user123' } }
+          const keyConfig = { KnownTable: new Set(['userId']) }
+
+          const result = DynamoDb.calculatePutItemHash(tableName, item, keyConfig)
+          expect(result).to.be.undefined
+        })
+
+        it('returns undefined for empty primary key config', () => {
+          const tableName = 'UserTable'
+          const item = { userId: { S: 'user123' } }
+
+          const result = DynamoDb.calculatePutItemHash(tableName, item, {})
+          expect(result).to.be.undefined
+        })
+
+        it('returns undefined for invalid primary key config', () => {
+          const tableName = 'UserTable'
+          const item = { userId: { S: 'user123' } }
+          const invalidConfig = { UserTable: ['userId'] } // Array instead of Set
+
+          const result = DynamoDb.calculatePutItemHash(tableName, item, invalidConfig)
+          expect(result).to.be.undefined
+        })
+
+        it('handles missing attributes in item', () => {
+          const tableName = 'UserTable'
+          const item = { someOtherField: { S: 'value' } }
+          const keyConfig = { UserTable: new Set(['userId']) }
+
+          const actualHash = DynamoDb.calculatePutItemHash(tableName, item, keyConfig)
+          const expectedHash = generatePointerHash([tableName, 'userId', Buffer.from(''), '', ''])
```

Comment:
i would actually expect this to be no hash at all, since we didn't have the key values.

Reply:
if we hit this a lot, we would fail toward very aggressively linked spans, which is probably the wrong way to go.

Reply:
Yes i agree, thanks for the suggestion